src/www/app/outline_server_repository/index.ts

```diff
@@ -187,9 +193,13 @@ export class OutlineServerRepository implements ServerRepository {
       throw new Error(`could not parse saved V0 servers: ${e.message}`);
     }
     for (const serverId of Object.keys(configById)) {
-      const config = configById[serverId];
+      const {name, ...config} = configById[serverId];
```

Comment:
This is a bit too magic. Could you assign to a configV0 object and explicitly extract the name? You'll need a new type name.

Also, it's hard to tell that removing the name from configv0 you get the ShadowsocksSessionConfig. That assumption is adding a hidden dependency between the types. I'd rather you create the ShadowsocksSessionConfig explicitly.

However, `shadowsocksSessionConfigToAccessKey` should change to take the v0 config, not the ShadowsocksSessionconfig. We go from accessKey to config, but don't need to go from config to access key, except here.

The function `shadowsocksSessionConfigToAccessKey` should probably be renamed and live in this file somewhere.

Reply:
You can also consider having a library for storage concerns (rather than serialization. Serialization is always tied to some purpose, better to group by the purpose)